src/main/java/net/tropicraft/core/common/entity/underdasea/atlantoku/EntityTropicraftWaterBase.java

```diff
@@ -614,4 +645,49 @@ public void setExpRate(int i) {
 	public void setHostile() {
 		this.canAggress = true;
 	}
+	
+	
+	
+
+	@Override
+	public void writeEntityToNBT(NBTTagCompound n) {
+		n.setString("texture", this.getDataManager().get(TEXTURE));
+		n.setFloat("swimYaw", this.getDataManager().get(SWIMYAW));
+		n.setFloat("swimPitch", this.getDataManager().get(SWIMPITCH));
+		super.writeEntityToNBT(n);
+	}
+
+	@Override
+	public void readEntityFromNBT(NBTTagCompound n) {
+		setTexture(n.getString("texture"));
+		this.swimYaw = n.getFloat("swimYaw");
+		this.swimPitch = n.getFloat("swimPitch");
+		this.getDataManager().set(SWIMYAW, this.swimYaw);
+		this.getDataManager().set(SWIMPITCH, n.getFloat("swimPitch"));
+		super.readEntityFromNBT(n);
+	}
+
+	
+	public void setTexture(String s) {
+		if(s.length() == 0) return;
+		if(!world.isRemote) {
+			this.getDataManager().set(TEXTURE, s);
+		}
+	}
+	
+	public void assignRandomTexture() {
+		if(!world.isRemote) {
+			if(getTexturePool().length > 0) {
```

Comment:
Since getTexturePool can return null, you should add a null check here.

Reply:
i could do, but for now it is only called if getTexturePool is not null